tidy3d/components/types.py

```diff
@@ -180,6 +180,8 @@ def __modify_schema__(cls, field_schema):
 
 Ax = Axes
 PlotVal = Literal["real", "imag", "abs"]
+FieldVal = Literal["real", "imag", "abs", "abs^2", "phase"]
```

Comment:
In order to maintain backwards compatibility (and in keeping with that warning you have right now), we may need to temporarily add `int` here? I think that pydantic might enforce type hints (@tylerflex correct me if I'm wrong), and my concern is that the plot functions won't accept `int` as a valid argument when a `FieldVal` is expected. If that's not true, then no worries.

Reply:
pydantic won't enforce it for methods of the class (wish it did!) so it should be ok.

Reply:
`'int'` is passed as a `field_name`, not `val`, so even if pydantic were to check this, it wouldn't be a problem, as `field_name` is a general `str`.